.github/workflows/workflows.yml

```
@@ -0,0 +1,432 @@
+          flags: pytest
+          verbose: true
+
+  test_ds003:
+    needs: [build, cache_test_data]
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        test_name: ["afni_smooth", "nistats_smooth", "afni_blurto", "nistats_blurto"]
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up docker buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: Download artifact
+        uses: actions/download-artifact@v2
+        with:
+          name: docker
+          path: /tmp
+
+      - name: Load docker image
+        run: |
+          docker info
+          set +o pipefail
+          ls -l /tmp/docker.tar
+          docker load < /tmp/docker.tar
+          docker images
+
+      - name: Restore cached test data
+        uses: actions/cache@v2
+        id: cache
+        with:
+          path: /tmp/.cache/data
+          key: ds003-v4-${{ runner.os }}
+          restore-keys: |
+            ds003-v4-${{ runner.os }}
+            ds003-v4-
+            ds003-
+
+      - name: Run fitlins
+        timeout-minutes: 240
+        env:
+          CONDA_PREFIX: /opt/miniconda-latest/envs/neuro
+        run: |
+          mkdir -p /tmp/ds003/work /tmp/ds003/derivatives
+          chmod 777 /tmp/ds003/work /tmp/ds003/derivatives
+          export CONDA_PREFIX=/opt/miniconda-latest/envs/neuro
+          echo $CONDA_PREFIX
+          docker run --rm -v /tmp/.cache/data/fitlins_tests:/data:ro \
+                -v /tmp/ds003/derivatives:/out \
+                -v /tmp/ds003/work:/scratch \
+                -v $GITHUB_WORKSPACE:/src/fitlins \
+                -e COVERAGE_FILE=/scratch/.coverage_${{ matrix.test_name }} \
+                --entrypoint="/neurodocker/startup.sh" \
+                ${{ github.repository }}:latest \
+                pytest --cov=fitlins \
+                --cov-config /src/fitlins/docker/multiproc.coveragerc \
+                --cov-report xml:/scratch/.coverage_pytest_${{ matrix.test_name }}.xml \
+                $CONDA_PREFIX/lib/python3.9/site-packages/fitlins/tests \
+                --fitlins-path=$CONDA_PREFIX/bin/fitlins \
+                --bids-dir=/data/ds003/fmriprep/sourcedata \
+                --output-dir=/out \
+                --derivatives=/data/ds003/fmriprep \
+                --model=/src/fitlins/examples/models/ds000003/models/model-001_smdl.json \
+                --work-dir=/scratch \
+                --test-name=${{ matrix.test_name }} \
+                --database-path=/out/ds003_database \
+                --reference-dir=/data/ds003
+
+      - name: Combine coverage and submit
+        uses: codecov/codecov-action@v2
+        with:
+          files: /tmp/ds003/work/.coverage_pytest_${{ matrix.test_name }}.xml
+          flags: ${{ matrix.test_name }}
+          env_vars: GITHUB_JOB
+          verbose: true
+
+  deploy_version:
+    needs: [build_docs, test_ds003]
+    if: startsWith(github.ref, 'refs/tags/v')
```

I think this would require us to tag with `vX.Y.Z`, which we don't currently do.

```suggestion
    if: startsWith(github.ref, 'refs/tags/')
```